src/oumi/core/analyze/dataset_analyzer.py

```
@@ -0,0 +1,129 @@
+# Copyright 2025 - Oumi
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+
+from oumi.core.configs import AnalyzeConfig
+from oumi.core.registry.registry import REGISTRY
+from oumi.utils.analysis_utils import (
+    compute_sample_level_analysis,
+    load_dataset_from_config,
+)
+from oumi.utils.logging import logger
+
+
+class DatasetAnalyzer:
+    """Orchestrates dataset analysis by creating and managing sample analyzers."""
+
+    def __init__(self, config: AnalyzeConfig):
+        """Initialize the dataset analyzer with configuration.
+
+        Args:
+            config: AnalyzeConfig object containing all analysis parameters
+        """
+        self.config = config
+        self.dataset_name = config.dataset_name
+        self.split = config.split
+
+        self.dataset = load_dataset_from_config(config)
+        self.sample_analyzers = self._initialize_sample_analyzers()
+
+    def _initialize_sample_analyzers(self):
+        """Initialize sample analyzer plugins from configuration."""
+        sample_analyzers = {}
+        for analyzer_config in self.config.analyzers:
+            try:
+                # Get the analyzer class from the registry
+                analyzer_class = REGISTRY.get_sample_analyzer(analyzer_config.id)
+                if analyzer_class is None:
```

Should we raise an exception here instead of proceeding?